pkg/reconciler/namespacecontroller/namespace_predicate.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespacecontroller
+
+import (
+	"k8s.io/klog/v2"
+	"kpt.dev/configsync/pkg/api/configmanagement"
+	"kpt.dev/configsync/pkg/syncer/differ"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var _ predicate.Predicate = NamespaceLabelChangedPredicate{}
+
+// NamespaceLabelChangedPredicate triggers a new reconciliation when Namespace
+// labels are changed, so NamespaceSelector can select a different set of resources.
+type NamespaceLabelChangedPredicate struct {
+	predicate.Funcs
+}
+
+// canTriggerReconcile returns whether the Namespace has labels that can trigger
```

Comment:
Is this function generic for any objects or just for namespaces? Suggest making it more clear, e.g. change the input type, do a type check

Reply:
The predicate is only used for Namespace watch, but it doesn't hurt to have a type check.
done